Update alert block for composer dependencies (#435)
PrestaShop/docs: src/content/1.7/modules/concepts/composer.md

**Don't forget to publish your vendor folder**

> Composer is responsible in downloading your dependencies and generating the autoloader from the `composer.json`

Reviewer: ```suggestion
Composer is responsible for downloading your dependencies and generating the autoloader from the `composer.json`
```

& `composer.lock` in your source code. These two artifacts are needed by your module to work properly, so before

> releasing it don't forget to run `composer dump-autoload -o --no-dev` to include your **production dependencies**

Reviewer 2: ```suggestion
releasing it don't forget to run `composer dump-autoload -o --no-dev` to only include your **production dependencies**
```

Author: There is something weird about this `only`. I wrote it at the beginning but decided to remove it, it's like we add the prod dependencies but we forget all the rest

Reviewer 2: ```suggestion
releasing it don't forget to run `composer dump-autoload -o --no-dev` to install your dependencies for a **production** environment
```
Maybe better? wdyt?

> in the archive.

Reviewer: I think this notice is confusing.

Here's my suggestion:

```
{{% notice warning %}}
**Don't forget to include your vendor folder in your release package**
Composer is responsible for downloading your dependencies and creating the autoloader file that your module will need to work properly. Therefore, remember to run `composer dump-autoload -o --no-dev` before you create your package, and make sure you include the `vendor` directory in it.
{{% /notice %}}

{{% notice warning %}}
**Don't include development dependencies in your release package**
Development-only libraries like PHPUnit can pose a security threat in production environments. Make sure that these libraries are imported as "dev dependencies" (`require-dev`) so that they aren't included when you create your release package. If in doubt, **double-check that they aren't  in the `vendor` directory before creating your release package**.
{{% /notice %}}
```

Author: > double-check that they aren't in the `vendor` directory before creating your release package

Should be `before submitting` or `when creating` don't you think?
